components/shopify/actions/create-product/create-product.mjs

```diff
@@ -0,0 +1,108 @@
+import shopify from "../../shopify.app.js";
+
+export default {
+  key: "shopify-create-product",
+  name: "Create Product",
+  description: "Create a new product",
+  version: "0.0.1",
+  type: "action",
+  props: {
+    shopify,
+    title: {
+      propDefinition: [
+        shopify,
+        "title",
+      ],
+    },
+    bodyHtml: {
+      type: "string",
+      label: "Body <HTML>",
+      description: `A description of the product
+        Supports HTML formatting
+        Example: \`<strong>Good snowboard!</strong>\``,
+      optional: true,
+    },
+    vendor: {
+      type: "string",
+      label: "Vendor",
+      description: "The name of the product's vendor",
+      optional: true,
+    },
+    productType: {
+      type: "string",
+      label: "Product Type",
+      description: "A categorization for the product used for filtering and searching products",
+      optional: true,
+    },
+    status: {
```

Comment:
Ok, I thought about this but was not sure what was the best approach. Let's suppose we see that a prop could be reused at some point but will not be in the moment. For the actions in the scope it is not reused.

Would it be better to put it in the app file or transferred only in the future when it would be reused?

Reply:
According to the [guidelines](https://pipedream.com/docs/components/guidelines/#prop-definitions), new app-specific props that may be reused in future components should be added to the app file.

Generally, I think that both approaches make sense. Defining props in the app file promotes reusability, but can make the app file less readable.